### PR TITLE
Allow Kanban titles to use referenced data

### LINKED ISSUE #23537: Ability to use a referenced collection's value in the kanban title/text field
### Describe the Improvement

At the minute it's only possible to see a referenced collection's ID which doesn't give the best experience.

### PR DESCRIPTION
**Pull request by matt-rolley**
## Scope

What's changed:

- Ability to use a referenced collections display value in the Kanban title

## Potential Risks / Drawbacks

- Could potentially add/load more related fields then required?

## Review Notes / Questions

- I've not done much reference field loading in the directus app before, so if there is a better way to do it please let me know :)
- Any review/advice would be greatly appreciated.

---

Fixes #23537

### CODE REVIEW

**Comment by hanneskuettner** [comment]
Hey @matt-rolley! Thank you for the PR! 

I made a couple of changes, that include fixing a regression, that would not allow normal text fields to be selected anymore and simplifying the addition of the fields necessary for rendering the display template.

**Review by connorwinston — DISMISSED** [review]
**Few notes:**

- Fields that have relations only show up if they have a display template set, they should be selectable regardless if they have a display configured with a display template.
- Is this supposed to add support for only `M2O` or also `O2M`, `M2M`, and `M2A`? Only `M2O` works for me.
- The styling needs some work, setting the title to be `user_created` for users who do not have a profile picture results in the default material icon being huge.
- When I initially switched from the table to kanban layout, the cards did not load and instead were replaced by a red error message saying it couldn't read `textField` of `undefined`. Deselecting the title field and reselecting the same field made the error go away and the cards to load. This error only showed up on the initial time the kanban was loaded so there may need to be some logic to update existing configs? _Not sure about this one._

**Comment by formfcw** [comment]
> * Fields that have relations only show up if they have a display template set, they should be selectable regardless if they have a display configured with a display template.

❗ I could not reproduce this. Is it still valid or can you provide more detailed steps to reproduce it?

> * Is this supposed to add support for only `M2O` or also `O2M`, `M2M`, and `M2A`? Only `M2O` works for me.

✅ Done

> * The styling needs some work, setting the title to be `user_created` for users who do not have a profile picture results in the default material icon being huge.

✅ Done

> * When I initially switched from the table to kanban layout, the cards did not load and instead were replaced by a red error message saying it couldn't read `textField` of `undefined`. Deselecting the title field and reselecting the same field made the error go away and the cards to load. This error only showed up on the initial time the kanban was loaded so there may need to be some logic to update existing configs? _Not sure about this one._

❗ I could not reproduce this. Is it still valid or can you provide more detailed steps to reproduce it?

---

From a functional point of view, it looks good to me, at the same time I would strongly recommend a design review of the card's appearance by the creative team, as it seems to have changed a lot over time.

File: app/src/layouts/kanban/kanban.vue [hunk 1]
```diff
@@ -88,6 +88,10 @@ function saveChanges() {
 	editTitle.value = '';
 }
 
+const titleFieldConfiguration = computed<Field | undefined>(() => {
+	return props.fieldsInCollection.find((field) => field.field === props.layoutOptions?.titleField);
+});
+
 const textFieldConfiguration = computed<Field | undefined>(() => {
 	return props.fieldsInCollection.find((field) => field.field === props.layoutOptions?.textField);
 });
```
Comment (AlexGaillard) [inline app/src/layouts/kanban/kanban.vue:97]: @formfcw thoughts on maybe combining these into a single computed prop, seeing as they're basically exactly the same?

```
const fieldConfigurations = computed<{ textField: Field | undefined; titleField: Field | undefined }>(() => {
	return {
		textField: props.fieldsInCollection.find((field) => field.field === props.layoutOptions?.textField),
		titleField: props.fieldsInCollection.find((field) => field.field === props.layoutOptions?.titleField),
	};
});
```

File: app/src/layouts/kanban/kanban.vue [hunk 2]
```diff
@@ -159,18 +163,33 @@ const textFieldConfiguration = computed<Field | undefined>(() => {
...
+												:collection="collection"
+												:value="element.text"
+												:type="textFieldConfiguration.type"
+												:field="layoutOptions?.textField"
+												:display="textFieldConfiguration.meta?.display"
+												:options="textFieldConfiguration.meta?.display_options"
+												:interface="textFieldConfiguration.meta?.interface"
+												:interface-options="textFieldConfiguration.meta?.options"
+											/>
+										</div>
```
Comment (AlexGaillard) [inline app/src/layouts/kanban/kanban.vue:192]: With the above in mind of combining the fieldConfigurations into a single computed var, then we could render both of these with a single v-for loop 🤔 

The  currently sits between them, but I think the CSS is such that we can move it below without affecting the layout.

**Review by AlexGaillard — CHANGES_REQUESTED** [review]
@formfcw I think this looks great and is a useful addition. Just a minor suggestion here on drying the code a bit. But if you're against it for any reason then please say so. 😄

File: app/src/layouts/kanban/kanban.vue (hunk 2 above)
Comment (formfcw) [reply app/src/layouts/kanban/kanban.vue]: I don’t think a for-loop makes sense here, at the same time I agree that the elements look the same. That’s why I refactored the computed property to pass the `<render-display>` options via `v-bind`.

**Review by AlexGaillard — APPROVED** [review]
Thanks for finishing this @formfcw, and thanks for the suggestion and work @matt-rolley. 👍 A useful addition.

**Pull request merged**